custom_components/mypyllant/climate.py

```diff
@@ -642,3 +666,174 @@ async def async_set_preset_mode(self, preset_mode):
+    )
+    _attr_preset_modes = AMBISENSE_ROOM_PRESETS
+    _enable_turn_on_off_backwards_compatibility = False
+
+    def __init__(
+        self,
+        system_index: int,
+        room_index: int,
+        coordinator: SystemCoordinator,
+        config: ConfigEntry,
+        data: dict,
+    ) -> None:
+        super().__init__(coordinator)
+        self.system_index = system_index
+        self.room_index = room_index
+        self.config = config
+        self.data = data
+        self.data["last_active_hvac_mode"] = (
+            self.hvac_mode if self.hvac_mode != HVACMode.OFF else HVACMode.AUTO
+        )
+        _LOGGER.debug(
+            "Saving last active HVAC mode %s", self.data["last_active_hvac_mode"]
+        )
+
+    async def async_update(self) -> None:
+        """
+        Save last active HVAC mode after update, so it can be restored in turn_on
+        """
+        await super().async_update()
+
+        if self.enabled and self.hvac_mode != HVACMode.OFF:
+            _LOGGER.debug("Saving last active HVAC mode %s", self.hvac_mode)
+            self.data["last_active_hvac_mode"] = self.hvac_mode
+
+    @property
+    def default_quick_veto_duration(self):
+        return self.config.options.get(
+            OPTION_DEFAULT_QUICK_VETO_DURATION, DEFAULT_QUICK_VETO_DURATION
+        )
+
+    @property
+    def system(self) -> System:
+        return self.coordinator.data[self.system_index]
+
+    @property
+    def room(self) -> AmbisenseRoom:
+        return [
+            r for r in self.system.ambisense_rooms if r.room_index == self.room_index
+        ][0]
+
+    @property
+    def id_infix(self) -> str:
+        return f"{self.system.id}_room_{self.room_index}"
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        return DeviceInfo(
+            identifiers={(DOMAIN, self.id_infix)},
+            name=self.name,
+            manufacturer=self.system.brand_name,
+        )
+
+    @property
+    def unique_id(self) -> str:
+        return f"{DOMAIN}_{self.id_infix}_climate"
+
+    @property
+    def name(self) -> str:
+        return f"{self.system.home.home_name or self.system.home.nomenclature} {self.room.name}"
+
+    @property
+    def extra_state_attributes(self) -> Mapping[str, Any] | None:
+        attr = {
+            "time_program": self.room.time_program,
+            "quick_veto_end_date_time": self.room.room_configuration.quick_veto_end_time,
+            "window_state": self.room.room_configuration.window_state,
+            "button_lock": self.room.room_configuration.button_lock,
+        }
+        return attr | self.room.extra_fields
+
+    async def set_quick_veto(self, **kwargs):
+        _LOGGER.debug("Setting quick veto on %s with params %s", self.room.name, kwargs)
+        temperature = kwargs.get("temperature")
+        duration_hours = kwargs.get("duration_hours")
+        await self.coordinator.api.quick_veto_ambisense_room(
```

Comment:
Ok, got the HTTP 500 error. API should return a bad request instead of 500, but anyway..

This is working fine when setting quick veto through the service with `duration_hours` an actual value in seconds. I can reproduce the error by calling 
`api.quick_veto_ambisense_room(room, 15, None, 3)` : 
1. `default_duration` is defined, so it stays `3`
2. `duration_seconds` is not defined, so it picks up `default_duration` (=3),
3. It breaks the API :innocent: 

IMO, solution should be:
```python
duration_seconds = duration_hours * 60 if duration_hours else None
await self.coordinator.api.quick_veto_ambisense_room(
    self.room, temperature, duration_seconds, self.default_quick_veto_duration
)
```
-> So the service is still usable as is. If we pass 3 in `duration_hours`, it will be transformed to seconds.

and (note the `* 60`, to get it in seconds)
```python
    @property
    def default_quick_veto_duration(self):
        return self.config.options.get(
            OPTION_DEFAULT_QUICK_VETO_DURATION, DEFAULT_QUICK_VETO_DURATION
        ) * 60
```

Reply:
I assume you meant to write `* 3600` for seconds in your examples?

Here's a shot at fixing both sections of code: https://github.com/signalkraft/mypyllant-component/pull/138/commits/e8ddb3dcd24f669c6d2823fccde12c865a615457

I added `duration_seconds` as a possible `kwargs` parameter, to be able to add it to the corresponding service later. Seems like the service should support the native unit if the underlying API (but passing in hours also works).

Reply:
ah my bad :disappointed:, quick veto for a room is expressed is in **minutes** , so *60 is correct, but variable names are wrong (even in the lib)

Reply:
No worries, here's one last quick shot at this for today: https://github.com/signalkraft/mypyllant-component/releases/tag/v0.8.0b2